tb_plugin/torch_tb_profiler/profiler/gpu_metrics_parser.py

```diff
@@ -77,19 +86,20 @@ def get_bucket_info(range_micro_seconds):
                         gpu_utilization_timeline[gpu_id][current_bucket_index] += (right_bound - left_bound)
                         if current_bucket[1] < current_range[1]:
                             current_bucket_index += 1
-                            current_bucket = (steps_start_time + current_bucket_index * bucket_size,
-                                              steps_start_time + (current_bucket_index + 1) * bucket_size)
+                            current_bucket = buckets_ranges[current_bucket_index] if current_bucket_index < buckets \
+                                else None
                         else:
                             current_range_index += 1
                             if current_range_index < len(self.kernel_ranges_per_device[gpu_id]):
                                 current_range = self.kernel_ranges_per_device[gpu_id][current_range_index]
                 for i_bucket in range(buckets):
```

Comment:
could the two loops be merged?

Reply:
I prefer uses 2 separate loop because the later one has some processing code out of loop. And the buckets are less than 100, little perf. overhead.

Reply:
it would be better that we can add some comments about the reason why uses two loops here.

Reply:
Well, merged into one loop.